Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3810 @@
+    if context.get("context"):
+        context = context.get("context")
+    username = params.get("credentials", {}).get("identifier", "")
+    password = params.get("credentials", {}).get("password", "")
+    client_id = params.get("client_id") or context.get("client_id")
+    client_secret = params.get("client_secret") or context.get("client_secret")
+    access_token = params.get("access_token") or context.get("access_token")
+    command = demisto.command()
+    demisto.debug(f"Command being called is {command}")
+
+    try:
+        # test_module functionality is disabled for Auth Token authentication
+        case_auth_token_auth = bool(
+            params.get("access_token", None) and not params.get("client_id", None) and not params.get("client_secret", None)
+        )
+        if command == "test-module" and not use_basic_auth and not case_auth_token_auth:
+            return_results(
+                test_module(
+                    client=None,
+                    use_basic_auth=use_basic_auth,
+                    client_id=params.get("client_id"),
+                    client_secret=params.get("client_secret"),
+                    access_token=params.get("access_token"),
+                    username=username,
+                    password=password,
+                )
+            )
+
+        auth, headers = create_client_header(use_basic_auth, username, password, client_id, client_secret, access_token)
+
+        client = Client(
+            server_url=url,
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy,
+            auth=auth,
+            organization_params=params.get("organization"),
+            client_id_param=params.get("client_id"),
+            client_secret_param=params.get("access_token"),
+        )
+
+        # check auth token validity and if a refresh token is needed to obtain new auth token
+        if not use_basic_auth:
+            if new_token := check_and_update_token(client, client_id, client_secret, context):
+                client._headers = {"Authorization": f"Bearer {new_token}"}
+
+        commands = {
+            "test-module": test_module,
+            "vd-auth-start": handle_auth_token_command,
+            "vd-auth-test": auth_test_command,
+            "vd-appliance-list": appliance_list_command,
+            "vd-organization-list": organization_list_command,
+            "vd-organization-appliance-list": appliances_list_by_organization_command,
+            "vd-appliance-group-list": appliances_group_list_by_organization_command,
+            "vd-appliance-group-template-appliance-list": appliances_list_by_device_group_command,
+            "vd-template-list": template_list_by_organization_command,
+            "vd-datastore-template-list": template_list_by_datastore_command,
+            "vd-application-service-template-list": application_service_template_list_command,
+            "vd-template-change-commit": template_change_commit_command,
```

Comment:
```suggestion
```

Reply:
?

Reply:
Same as test-module - it is called separately.

Reply:
done